src/goto-programs/goto_trace.cpp

```diff
@@ -348,6 +348,145 @@ bool is_index_member_symbol(const exprt &src)
     return false;
 }
 
+/// \brief show a compact variant of the goto trace on the console
+/// \param out the output stream
+/// \param ns the namespace
+/// \param goto_trace the trace to be shown
+/// \param options any options, e.g., numerical representation
+void show_compact_goto_trace(
+  messaget::mstreamt &out,
+  const namespacet &ns,
+  const goto_tracet &goto_trace,
+  const trace_optionst &options)
+{
+//  unsigned prev_step_nr=0;
+//  bool first_step=true;
+  std::size_t function_depth=0;
+
+  for(const auto &step : goto_trace.steps)
+  {
+    if(step.is_function_call())
+      function_depth++;
+    else if(step.is_function_return())
+      function_depth--;
+  
+    // hide the hidden ones
+    if(step.hidden)
+      continue;
+
+    switch(step.type)
+    {
+    case goto_trace_stept::typet::ASSERT:
+      if(!step.cond_value)
+      {
+        out << '\n';
+        out << messaget::red << "Violated property:" << messaget::reset << '\n';
+        if(!step.pc->source_location.is_nil())
+        {
+          out << "  " << state_location(step, ns) << '\n';
+        }
+
+        out << "  " << messaget::red << step.comment << messaget::reset << '\n';
+
+        if(step.pc->is_assert())
+          out << "  " << from_expr(ns, step.function, step.pc->guard) << '\n';
+
+        out << '\n';
+      }
+      break;
+
+    case goto_trace_stept::typet::ASSIGNMENT:
+      if(step.assignment_type == goto_trace_stept::assignment_typet::ACTUAL_PARAMETER)
+        break;
+    
+      out << "  ";
+
+      if(!step.pc->source_location.get_line().empty())
+      {
+        out << messaget::faint
+            << step.pc->source_location.get_line()
+            << ':' << messaget::reset << ' ';
+      }
+
+      trace_value(
+        out,
+        ns,
+        step.get_lhs_object(),
+        step.full_lhs,
+        step.full_lhs_value,
+        options);
+      break;
+
+    case goto_trace_stept::typet::FUNCTION_CALL:
+      // downwards arrow
+      out << "\n" << messaget::faint << u8"\u21b3" << messaget::reset << ' ';
+      if(!step.pc->source_location.get_file().empty())
+      {
+        out << messaget::faint
+            << step.pc->source_location.get_file();
+
+        if(!step.pc->source_location.get_line().empty())
+        {
+          out << messaget::faint << ':'
+              << step.pc->source_location.get_line();
+        }
+        
+        out << messaget::reset << ' ';
+      }
+
+      {
+        // show pretty name until first '('
+        const auto &f_symbol = ns.lookup(step.called_function);
+        const auto display_name = id2string(f_symbol.display_name());
+        const std::size_t open_par_pos = display_name.find('(');
```

Comment:
What's the idea behind this filtering?

Reply:
A pretty name might be some_function(int, int); we want to show some_function(123, 456)

Reply:
Why would the symbol's name include the type of the arguments? Is that in case of templates? But then `operator()` would not be handled correctly. I feel I'm missing something.

Reply:
Ok, will drop for now